datacube_alchemist/_utils.py

```diff
@@ -124,6 +124,8 @@ def _munge_dataset_to_eo3(ds: Dataset) -> DatasetDoc:
     product = ProductDoc(name=ds.type.name)
     # Wrap properties to avoid typos and the like
     properties = StacPropertyView(ds.metadata_doc.get("properties", {}))
+    if properties.get('eo:gsd'):
```

Comment:
```python
properties.pop('eo:gsd', None)
```
achieves the same in one line and without possible concurrency issues, not that they matter here.

Reply:
That doesn't work, for some reason. I guess because it's some object and not a dict.

Reply:
then maybe we should do it on a dict and only then wrap it in an object, but whatever... see my other comment